src/lightning/fabric/fabric.py

```diff
@@ -1063,3 +1062,14 @@ def _configure_callbacks(callbacks: Optional[Union[List[Any], Any]]) -> List[Any
         callbacks = callbacks if isinstance(callbacks, list) else [callbacks]
         callbacks.extend(_load_external_callbacks("lightning.fabric.callbacks_factory"))
         return callbacks
+
+
+def _update_device_attribute(module: torch.nn.Module) -> None:
+    # this assumes that the module is already on the desired device
+    if not isinstance(module, _DeviceDtypeModuleMixin):
+        return
```

Comment:
This early exit is to avoid leaking a `_device` attribute if this was called with something else. I can remove it if you prefer